<link href="https://fonts.googleapis.com/css?family=Roboto:100,300,400" rel="stylesheet">

<link rel="stylesheet" href="about.css">

<link rel="stylesheet" href="/tippy/tippy.css">
<script src="/tippy/tippy.min.js"></script>

  <div class="content_center">

    <table border=0>
      <tr>
        <td class="content_table" valign=top>
          <div class="div_content"><b>expressRNA</b> is an umbrella term for the web application and platform to study post-transcriptional modifications. expressRNA builds upon several novel open-source bioinformatics packages (<a href=https://github.com/meringlab/apa target=_apa>apa</a>, <a href=https://github.com/grexor/rnamotifs2 target=_rna2>RNAmotifs2</a> and <a href=https://github.com/grexor/pybio target=_pybio>pybio</a>) to perform <b>alternative polyadenylation</b> and <b>alternative splicing</b> analysis. The platform also inter-connects several state-of-the-art tools like the short-read STAR aligner, DEXSeq for differential usage estimates and minimap2 for long-read alignment.</div>

          <br>

          <div class="div_header"><b>Latest News</b>, <a href="javascript:open_info()">see all news</a></div>

          <div class="div_content">
            <b><font color=#555555>22. February 2019</font></b><br>
            <div style="padding-left:8px; padding-top:5px;" align="justify">
              <img src="media/cloud.png" align="right">
              The expressRNA project is moving forward to provide <b>automatic data processing</b>.
              We plan to open test experiment data upload by end of February 2019. In this first stage, users will be able to upload experiments (FASTQ/FASTA) files and expressRNA will compute the polyA database, polyA expression profiles from the experimental data and additional global gene expression levels.
              <br><br>
              In terms of infrastructure, we are using a cloud server with 8-core CPU and 64 GB RAM, with enough disk space for this initial test of the data upload (scroll down to see <b>live server CPU usage, CLOUD</b>).
              <br><br>
              If you are logged into the system, you can notice an <a href='javascript:open_profile()'>updated profile section</a>, where you can see currently processed tasks / tickets.
              <br><br>
            </div>
          </div>

          <br>

          <div class="div_header"><b>Alternative Polyadenylation (APA) Computational Analysis</b></div>
          <div class="div_content">
            expressRNA implements <b>alternative polyAdenylation (APA)</b> analysis from 3'-end short-read targeted sequence data and long-read Nanopore / PacBio data. In general the analysis is flexible to accomodate diverse sequence data, for example <a href="https://www.lexogen.com/quantseq-3mrna-sequencing/?gclid=CjwKCAjwhOvPBRBxEiwAx2nhLqPMeeMzqnWVmvDLAR3uSNW-YITTbG0UHBWCkiZk1w3wVdzd2ThL_BoCYCMQAvD_BwE" target=_lexogen>Lexogen Quantseq Forward and Reverse</a> protocols are well supported.

            The following steps are crucial for APA analysis:

            <div style="padding-left: 15px; padding-top: 10px;">
              <li> map / align the reads to the reference with STAR short read aligner
              <li> construct a polyA atlas / database by clustering alignment ends (or beginnings, depending on the protocol)
              <li> filter out alignments from A-rich genomic regions (to avoid internal priming events in the atlas construction step)
              <li> recover the signal by computing the "expression" of polyA sites by clustering
              <li> define comparative analysis (set of control experiments vs. set of test experiments)
              <li> run DEXSeq analysis on the polyA sites
              <li> select two sites per gene to classify polyA events (shortening, lenghtening of genes)
              <li> perform motif analysis around regulated sites with RNAmotifs2
              <li> Gene Ontology Enrichment Analysis on APA regulated genes
            </div>

            <img src="media/analysis_steps.png" width=700px>

            The above list is just a set of automated analysis that can be performed on a set of experiments. Of course several alternatives can be implemented, such as using a pre-existing polyA atlas (database) for the analysis and computing the polyA site usage (counts) on the provided polyA site atlas.
            <br><br>
            Part of an APA report is a graphical representation of gene shortening and lenghtening. Our case study was an experiment of HEK-293 control vs. <font color=green class="btn" title="TDP-43 is a DNA and RNA-binding protein.<br><br><a target=_tdp43 href=https://en.wikipedia.org/wiki/TARDBP>Open wiki page</a> on TDP-43 to start exploring more.">TDP-43</font> KD. Repressed genes are blue (proximal polyA site is less used compared to the distal), and vice-versa for the enhanced genes. <a href="javascript:open_analysis('paper_final_all', 'High-resolution RNA maps demonstrate common principles of splicing and polyadenylation regulation by TDP-43');">Open the analysis here</a>.
            <br><br>
            expressRNA provides the basic analysis in a completelly automatized fashion, however for research projects a lot of custom work still needs to be performed in terms of data quality control and pre-processing. Currently we are working to automatize all steps of analysis.

          </div>

          <br>

          <div class="div_header"><b>Alternative Splicing Computational Analysis</b></div>
          <div class="div_content">
            expressRNA also implements computational analysis of alternative splicing in terms of calling DEXSeq to identify regulated exons and then calling RNAmotifs2 on the sets of regulated exons for detecting regulatory short-motifs. Additionally, RNAmotifs2 is a completelly independent package which can be also called separatelly.
          </div>

          <br>

          <div class="div_header"><b>expressRNA is free?</b></div>
          <div class="div_content">
            The software packages that expressRNA builds upon are <b>free for non-commercial use</b>.

            <br><br>

            <b>Academic setting:</b> If you are part of a research group and would like to run APA analysis on our server, please <a href="mailto:expressRNA@gmail.com">contact us</a> to establish a research collaboration. We can then either help you (time permitting) to setup expressRNA on your server or suggest performing APA analysis on expressRNA.org directly. We are excited about new datasets and potential novel solutions developed during research collaborations. We are all scientist at expressRNA and are constantly improving the platform so new research collaborations are always welcome.

            <br><br>

            <b>Commercial setting:</b> If you would like the team at expressRNA to help you with data analysis on expressRNA.org in a commercial setting, please <a href="mailto:expressRNA@gmail.com">contact us</a> for an offer. Your contribution will cover time and infrastructure (disk space, CPU). There is also a possibility for you to apply expressRNA in a commercial setting on your own, for that you would need a <a href="javascript:open_licence()">commercial licence</a> for the entire platform.

          </div>

          <br>

          <div class="div_header"><b>Architecture</b></div>
          <div class="div_content">
          <b>expressRNA</b> combines and integrates several existing software tools and proposes data integrative approaches
          for better understanding the underlying processes of post-transcriptioanl modifications. The web exploratory interface facilitates the browsing and interpretation of results.
          <br><br>
          <center>
          <img width=500px src="media/architecture.png?nocache=" + nocache>
          </center>
          </div>

          <br>

          <div class="div_header"><b>Citing expressRNA</b></div>
            <div class="div_content">
            <div style="padding-right: 10px;" id="paper_div"><a href="http://www.cell.com/cell-reports/abstract/S2211-1247(17)30522-3" target="_paper"><b>High-resolution RNA maps suggest common principles of splicing and polyadenylation regulation by TDP-43</b></a>
            <br>Rot, G., Wang, Z., Huppertz, I., Modic, M., Lenƒçe, T., Hallegger, M., Haberman, N., Curk, T., von Mering, C., Ule, J.
            <br>Cell Reports , Volume 19 , Issue 5 , 1056 - 1067
            </div>
          </div>

          <br>

          <div class="div_header"><b>Contact</b></div>
          <div class="div_content">
          Collaborations, be it reseach or commercial, ideas and suggestions are always welcome: please see <a href="http://www.expressrna.org/index.html?action=info&section=contact">Contacts</a>.
        </td>
        <td width=260px class="menu_table" valign=top>
          <table border=0>
            <tr><td colspan=3><div style="font-size: 13px;">GitHub repositories</div></td></tr>
            <tr>
              <td align=center><div class="titlegit_font"><img src="media/GitHub-Mark-32px.png" height=13px>&nbsp;<a href="https://github.com/meringlab/apa" target="_apa_github">apa</a></div></td>
              <td align=center><div class="titlegit_font"><img src="media/GitHub-Mark-32px.png" height=13px>&nbsp;<a href="https://github.com/grexor/rnamotifs2" target="_rnamotifs2_github">RNAmotifs2</a></div></td>
              <td align=center><div class="titlegit_font"><img src="media/GitHub-Mark-32px.png" height=13px>&nbsp;<a href="https://github.com/grexor/pybio" target="_pybio_github">pybio</a></div></td>
            </tr>
          </table>
          <br>
          <div class="div_git" id="lbl_commits">
          </div>
        </td>
      </tr>
    </table>

</div>

<script src="about.js"></script>
