<link href="https://fonts.googleapis.com/css?family=Roboto:100,300,400" rel="stylesheet">

<div class="container">

    <table border=0>
      <tr>
        <td class="content_table" valign=top>
          <div class="div_content"><b>expressRNA</b>: web research platform to study post-transcriptional modifications by analyzing data from diverse RNA sequencing protocols (RNA-seq, long-read or 3'-end targeted sequencing) and estimating <font color=green class="etippy" title="Polyadenylation is the addition of a poly(A) tail to a messenger RNA, canonically at 3'-UTR end. <a target=_new href=https://en.wikipedia.org/wiki/Polyadenylation#Alternative_polyadenylation>Alternative polyadenylation</a> is the process of adding the polyA tail at an alternative locus of the RNA molecule.">alternative polyadenylation (APA)</font> and <font color=green class="etippy" title="<a href=https://en.wikipedia.org/wiki/Alternative_splicing target=_new>Alternative splicing</a>, or differential splicing, is a regulated process during gene expression that results in a single gene coding for multiple proteins. In the process, exons can be included or excluded from the final transcript.">alternative splicing</font> regulation. This regulatory information is then combined with RNA-protein binding <font color=green class="etippy" title="<a href=https://en.wikipedia.org/wiki/CLIP target=_new>CLIP</a> (cross-linking immunoprecipitation) is a method used in molecular biology that combines UV cross-linking with immunoprecipitation in order to analyse protein interactions with RNA">CLIP</font> to study regulatory principles via RNA maps.

          <br><br>
          expressRNA builds upon 3 open-source bioinformatics packages (<a href='https://github.com/grexor/apa' target=_new>apa</a>, <a href='https://github.com/grexor/pybio' target=_new>pybio</a>, <a href='https://github.com/grexor/rnamotifs2' target=_new>RNAmotifs2</a>; see GitHub repositories) to perform 3'-end and other sequence data processing and runs diverse state-of-the-art tools (STAR aligner, DEXSeq, minimap2) to compute <a href="javascript:open_help()">various steps of the analysis</a>.
          
          <br><br>

          Moreover, the page you are currently visiting is a web-app written in JQuery and Javascript with a python (mod-wsgy) backend, intended for facilitating the sharing and exploration of results of analyses. The whole source code (front-end and back-end) is available as open-source at the <a href='https://github.com/grexor/expressrna' target=_new>expressRNA GitHub repository</a>.

          <br><br>

          <a href="javascript:open_analysis('paper_final_all');"><img src="media/rnamaps.png" width=60 align="left" style='padding-right: 5px; padding-top: 5px;'></a>Regarding APA analyses: after <div style='opacity: 0.5; border-radius: 5px; padding-left: 3px; padding-right: 3px; display: inline; background-color: #ff6961; color: white;'>enhanced</div> and <div style='opacity: 0.5; border-radius: 5px; padding-left: 3px; padding-right: 3px; display: inline; background-color: #6169ff; color: white;'>repressed</div> polyA sites and exons are determined in the experiment, the analysis integrates CLIP data to generate RNA maps. These maps display the level of binding of a protein of interest around regulated features (polyA sites or exons) in a cumulative way across several genes.

          <br><br>
          <img src="media/rnamotifs.png" width=60 align="left" style='padding-right: 5px; padding-top: 5px;'>Additionally, we run motif analysis software RNAmotifs2, originally published in Cereda et al. 2014 and extended with cluster motif analysis in Rot et al., 2017. Detected motifs and their positional significance is evaluated and plotted against the same detected regulated features (polyA sites and exons) as with the CLIP <a href="javascript:open_analysis('paper_final_all');">RNA maps</a>.

          </div>

          <br>

          <div class="div_header"><b>Latest News</b>, <a href="javascript:open_news_blog()">see all news</a></div>

          <b><font color=#555555>09. July 2021</font></b><br>
          The entire expressRNA project (including source code of web pages) is now accessible over the <a href='https://github.com/grexor/expressrna' target=_new>expressRNA public GitHub repository</a>. The development branch (devel) is intended for feature development and is regularly merged with the master branch.
          <br><br>
          If you find expressRNA software / platform useful for your research and data analysis, consider sponsoring it over the <a href='https://github.com/sponsors/grexor?o=sd&sc=t' target=_new>GitHub sponsors programme</a>.

          <br><br>

          <div class="div_header"><b>Alternative Polyadenylation (APA) Computational Analysis</b></div>
          <div class="div_content">
            expressRNA focus is on sequence data analysis and alternative polyadenylation analysis (see more in <a href='javascript:open_help();'>Docs</a>).
            Alternative polyadenylation switches are reported (one per gene) and a full report of regulated polyA sites is generated, with interactive plots to browse the results. Our case study was an experiment of HEK-293 with <font color=green class="etippy" title="TDP-43 is a DNA and RNA-binding protein.<br><br><a target=_tdp43 href=https://en.wikipedia.org/wiki/TARDBP>Open wiki page</a> on TDP-43 to start exploring more.">TDP-43</font> knock-down. <a href="javascript:open_analysis('paper_final_all');">Check out the analysis results</a> and the <a href="http://www.cell.com/cell-reports/abstract/S2211-1247(17)30522-3" target="_paper">publication</a>.
            <br><br>
            <center>
            <img width=340px src="media/apa_pipeline.png?nocache=" + nocache>
          </div>

          <br>

          <div class="div_header"><b>Architecture</b></div>
          <div class="div_content">
          expressRNA combines and integrates several existing software tools and proposes data integrative approaches
          for better understanding the underlying processes of post-transcriptioanl modifications. The web exploratory interface facilitates the browsing and interpretation of results.
          <br><br>
          <center>
          <img width=540px src="media/architecture.png?nocache=" + nocache>
          </center>
          </div>

          <br>

          <div class="div_header"><b>Citing expressRNA</b></div>
            <div class="div_content">
            <div style="padding-right: 10px;" id="paper_div"><a href="http://www.cell.com/cell-reports/abstract/S2211-1247(17)30522-3" target="_paper">High-resolution RNA maps suggest common principles of splicing and polyadenylation regulation by TDP-43</a>
            <br>Rot, G., Wang, Z., Huppertz, I., Modic, M., Lenƒçe, T., Hallegger, M., Haberman, N., Curk, T., von Mering, C., Ule, J.
            <br>Cell Reports , Volume 19 , Issue 5 , 1056 - 1067
            </div>
          </div>

          <br>

          <div class="div_header"><b>Contact</b></div>
          <div class="div_content">
          Automatic pipelines provide a basic overview of novel data, however research work usually requires additional steps and analysis. Please don't hesitate to <a href='mailto:gregor.rot@gmail.com?subject=collaboration'>contact us</a> in case you would like to start a collaboration. If you have a suggestion or discovered a problem in the analysis, you can use directly the <a href='https://github.com/grexor/expressRNA/issues' target=_new>expressRNA GitHub Issues</a>.
        </td>
        <td width=260px class="menu_table" valign=top>
          <table border=0>
            <tr><td colspan=3><div style="font-size: 13px;"><b>GitHub Repositories</b></div></td></tr>
            <tr>
              <td align=center><div class="titlegit_font"><img src="media/GitHub-Mark-32px.png" height=11px>&nbsp;<a href="https://github.com/grexor/expressrna" target="_expressrna_github">expressRNA</a></div></td>
              <td align=center><div class="titlegit_font"><img src="media/GitHub-Mark-32px.png" height=11px>&nbsp;<a href="https://github.com/grexor/apa" target="_apa_github">apa</a></div></td>
              <td align=center><div class="titlegit_font"><img src="media/GitHub-Mark-32px.png" height=11px>&nbsp;<a href="https://github.com/grexor/rnamotifs2" target="_rnamotifs2_github">RNAmotifs2</a></div></td>
              <td align=center><div class="titlegit_font"><img src="media/GitHub-Mark-32px.png" height=11px>&nbsp;<a href="https://github.com/grexor/pybio" target="_pybio_github">pybio</a></div></td>
            </tr>
          </table>
          <br>
          <div class="div_git" id="lbl_commits">
          </div>
        </td>
      </tr>
    </table>

<script src="about.js"></script>

</div>
