<link rel="stylesheet" href="info.css">

<div class="container">

  <table border=0 style="padding-left: 0px; padding-right:10px; margin-top: -2px; padding-top: 0px; padding-bottom: 10px;">
    <tr>
      <td colspan=2 class="content_table" valign=top style="padding-bottom: 10px;"><div class="div_header" style=><b>Contributors</b></div></td>
    </tr>
    <tr>
      <td class="content_table" valign=top width=60><img src="media/gregor.jpg" style="height:50; opacity: 0.8;"></td>
      <td class="content_table" valign=top style="text-align: justify"><a target=_gregor href="https://scholar.google.ch/citations?hl=en&user=Mzz8rBAAAAAJ&view_op=list_works&sortby=pubdate"><b>Gregor Rot</b></a><br>
        Gregor is the main developer and coder at expressRNA. Gregor started working on iCLIP and alternative polyadenylation analysis with Tomaž Curk and Jernej Ule in 2009 and obtained his Phd with Christian von Mering from the University of Zurich in 2016. Gregor currently holds an Embedded Bioinformatician position at the University of Zurich and is collaborating with several research laboratories around the world, mostly on computational analysis of post-transcriptional modifications.
      </td>
    </tr>
    <tr><td style="height:10px;"></td></tr>
    <tr>
      <td class="content_table" valign=top width=60><img src="media/curk.jpg" style="height:50; opacity: 0.8;"></td>
      <td class="content_table" valign=top style="text-align: justify"><a target=_tomaz href="https://scholar.google.com/citations?hl=en&user=0QCwhhYAAAAJ&view_op=list_works&sortby=pubdate"><b>Tomaž Curk</b></a><br>
        Tomaž developed the first version of the computational analysis pipeline of 3'-end targeted sequencing data. The data was obtained by applying the wet-lab protocol that Zhen Wang developed in Ule's lab. Tomaž continues to be involved in developing novel analysis methods of RNA-protein binding and post-transcriptional regulation and is Assistant Professor at the Faculty of Computer and Information Science, University of Ljubljana, Slovenia.
      </td>
    </tr>
    <tr><td style="height:10px;"></td></tr>
    <tr>
      <td class="content_table" valign=top width=60><img src="media/joao.jpg" style="height:50; opacity: 0.8;"></td>
      <td class="content_table" valign=top style="text-align: justify"><a target=_joao href="https://scholar.google.ch/citations?hl=en&user=HJAoBnkAAAAJ&view_op=list_works&sortby=pubdate"><b>João F. Matias Rodrigues</b></a><br>
        Joao is helping with several ideas and suggestions on computational data analysis and statistics, among others on integrating regulatory signals to RBPs via RNA-maps.
      </td>
    </tr>
    <tr><td style="height:10px;"></td></tr>
    <tr>
      <td class="content_table" valign=top width=60><img src="media/modic.jpg" style="height:50; opacity: 0.8;"></td>
      <td class="content_table" valign=top style="text-align: justify"><a target=_miha href="https://scholar.google.com/citations?hl=en&user=Quocg7EAAAAJ&view_op=list_works&sortby=pubdate"><b>Miha Modic</b></a><br>
        Miha is working with embryonic stem cells and studies the role of alternative polyadenylation in the context of cell differentiation and pluripotency, among other things. Miha contributed several datasets to expressRNA and suggested several improvements to the platform.
      </td>
    </tr>
    <tr><td style="height:10px;"></td></tr>
    <tr>
      <td class="content_table" valign=top width=60><img src="media/hallegger.jpg" style="height:50; opacity: 0.8;"></td>
      <td class="content_table" valign=top style="text-align: justify"><a target=_martina href="https://www.researchgate.net/profile/Martina_Hallegger"><b>Martina Hallegger</b></a><br>
        Martina performed the iCLIP experiments of TDP-43 in HEK-293 cells, which provided the crucial dataset to compute RNA-maps of TDP-43, disclosing the regulatory function of the protein in alternative polyadenylation. Martina continues to perform experimental work and use expressRNA for the study of 3'-end processing.
      </td>
    </tr>
    <tr><td style="height:10px;"></td></tr>
    <tr>
      <td class="content_table" valign=top width=60><img src="media/zhen.jpg" style="height:50; opacity: 0.8;"></td>
      <td class="content_table" valign=top style="text-align: justify"><a target=_zhen href="https://www.researchgate.net/profile/Zhen_Wang88"><b>Zhen Wang</b></a><br>
        Zhen developed a 3'-end targeted sequencing protocol during her Phd and PostDoc in Jernej Ule's lab, among studying the role of TIA in splicing and alternative polyadenylation. Zhen is now working for a startup company in Paris.
      </td>
    </tr>
    <tr><td style="height:10px;"></td></tr>
    <tr>
      <td class="content_table" valign=top width=60><img src="media/ina.jpg" style="height:50; opacity: 0.8;"></td>
      <td class="content_table" valign=top style="text-align: justify"><a target=_ina href="http://www.hentze.embl.de/ina-huppertz/"><b>Ina Huppertz</b></a><br>
        Ina performed the research work of alternative polyadenylation in HEK-293 cells in the context of TDP-43 knockdown. Ina's data provided one of the firts datasets to be analyzed with expressRNA and helped test and fine-tune the platform.
      </td>
    </tr>
    <tr><td style="height:10px;"></td></tr>
    <tr>
      <td class="content_table" valign=top width=60><img src="media/tina.jpg" style="height:50; opacity: 0.8;"></td>
      <td class="content_table" valign=top style="text-align: justify"><a target=_tina href="https://scholar.google.com/citations?hl=en&user=_1mBfGIAAAAJ&view_op=list_works&sortby=pubdate"><b>Tina Lenče</b></a><br>
        Tina helped with initial experimental work and data analysis on the study of TDP-43 in alternative polyadelylation. Tina graduated recently and continues to do excellent scientific research.
      </td>
    </tr>
    <tr><td style="height:10px;"></td></tr>
    <tr>
      <td class="content_table" valign=top width=60><img src="media/haberman.jpg" style="height:50; opacity: 0.8;"></td>
      <td class="content_table" valign=top style="text-align: justify"><a target=_nejc href="https://www.researchgate.net/profile/Nejc_Haberman"><b>Nejc Haberman</b></a><br>
        Nejc performed initial motif analysis for TDP-43 iCLIP which helped study the role of TDP-43 in alternative polyadenylation. Nejc is Postdoc at ICL, London.
      </td>
    </tr>

  </table>

  <br><br>

  <table border=0 style="padding-left: 0px; padding-right:10px; margin-top: -2px; padding-top: 5px; padding-bottom: 10px;">
    <tr>
      <td colspan=2 class="content_table" valign=top style="padding-bottom: 10px;"><div class="div_header" style=><b>Affiliated Research Groups</b></div></td>
    </tr>
    <tr>
      <td class="content_table" valign=top width=60><img src="media/3453453_23.jpg" style="height:50; opacity: 0.8;"></td>
      <td class="content_table" valign=top style="text-align: justify"><a target=_ueli href="https://www.botinst.uzh.ch/en/research/development/grossnik/research.html"><b>Ueli Grossniklaus Laboratory, Department of Plant and Microbial Biology, University of Zurich, Switzerland</b></a><br>
        Ueli Grossniklaus lab applies expressRNA in analysis of RNA-seq data and guides the development of methods for explorative analysis of gene expression and alternative polyadenylation.
      </td>
    </tr>
    <tr><td style="height:10px;"></td></tr>
    <tr>
      <td class="content_table" valign=top width=60><img src="media/j.jpg" style="height:50; opacity: 0.8;"></td>
      <td class="content_table" valign=top style="text-align: justify"><a target=_jernej href="http://ulelab.info"><b>Jernej Ule Laboratory, The Francis Crick Institute, London, UK</b></a><br>
        Jernej Ule lab is guiding research on APA and helping develop all aspects of expressRNA, including RNAmotifs2 analysis software. Jernej holds a joint position between UCL and The Francis Crick Institute, London, UK.
      </td>
    </tr>
    <tr><td style="height:10px;"></td></tr>
    <tr>
      <td class="content_table" valign=top width=60><img src="media/robinson.jpg" style="height:50; opacity: 0.8;"></td>
      <td class="content_table" valign=top style="text-align: justify"><a target=_robinson href="http://www.imls.uzh.ch/en/research/robinson.html"><b>Mark Robinson Laboratory, Institute of Molecular Life Sciences, Zurich, Switzerland</b></a><br>
        Mark Robinson lab is the main guide and advisor in analysis of long-read Nanopore and PacBio sequence data. There is a new wave of exciting opportunities arising from the long-read sequences spanning entire transcripts that expressRNA is trying to catch.
      </td>
    </tr>
    <tr><td style="height:10px;"></td></tr>
    <tr>
      <td class="content_table" valign=top width=60><img src="media/christian.jpg" style="height:50; opacity: 0.8;"></td>
      <td class="content_table" valign=top style="text-align: justify"><a target=_christian href="https://www.imls.uzh.ch/en/research/vonmering/people.html"><b>Christian von Mering Laboratory, Institute of Molecular Life Sciences, Zurich, Switzerland</b></a><br>
        Christian von Mering lab provided the initial excellent research environment where expressRNA was developed. Several lab members contributed ideas and suggestions.
      </td>
    </tr>
    <tr><td style="height:10px;"></td></tr>
    <tr>
      <td class="content_table" valign=top width=60><img src="media/ryan.jpg" style="height:50; opacity: 0.8;"></td>
      <td class="content_table" valign=top style="text-align: justify"><a target=_ryan href="https://ccr.cancer.gov/Laboratory-of-Human-Carcinogenesis/brid-m-ryan"><b>Brid Ryan Laboratory, National Cancer Institute, Washington, USA</b></a><br>
        Brid Ryan lab is using expressRNA to perform analysis of alternative polyadenylation in cancer since beginning of 2017. Brid and her team help develop expressRNA greatly by providing ideas and suggestions.
      </td>
    </tr>
    <tr><td style="height:10px;"></td></tr>
    <tr>
      <td class="content_table" valign=top width=60><img src="media/hornstein.jpg" style="height:50; opacity: 0.8;"></td>
      <td class="content_table" valign=top style="text-align: justify"><a target=_hornstein href="http://www.weizmann.ac.il/molgen/hornstein/"><b>Eran Hornstein Laboratory, Weizmann Institute of Science, Israel</b></a><br>
        Eran Hornstein lab is using expressRNA to study alternative polyadentlation in several mouse experiments connected with ALS. Eran and his team provide ideas and point out advantages and disadvantages of several expressRNA analysis. This is incredibly valuable information for the team at expressRNA to further improve and develop the platform.
      </td>
    </tr>
    <tr><td style="height:10px;"></td></tr>
    <tr>
      <td class="content_table" valign=top width=60><img src="media/fratta.jpg" style="height:50; opacity: 0.8;"></td>
      <td class="content_table" valign=top style="text-align: justify"><a target=_fratta href="http://www.ucl.ac.uk/ion/departments/sobell/Research/PFratta/People"><b>Pietro Fratta Laboratory, UCL London, UK</b></a><br>
        Pietro Fratta lab is applying expressRNA to libraries of 3'-end targeted and classic RNA-seq. Pietro and his team provide valuable input in terms of identifiying strenghts and weaknesses of expressRNA and help test the platform.
      </td>
    </tr>
    <tr><td style="height:10px;"></td></tr>
    <tr>
      <td class="content_table" valign=top width=60><img src="media/lehner.jpg" style="height:50; opacity: 0.8;"></td>
      <td class="content_table" valign=top style="text-align: justify"><a target=_lehner href="http://www.imls.uzh.ch/en/research/Lehner.html"><b>Christian Lehner Laboratory, Institute of Molecular Life Sciences, Zurich, Switzerland</b></a><br>
        Christian Lehner lab is studying alternative polyadenylation in Drosophila, with a very interesting experimental setup that is challenging in terms of comparative analysis. The research provides lot's of developmental opportunities for expressRNA.
      </td>
    </tr>
  </table>

</div>
