<link rel="stylesheet" href="help.css">

<div class="content_center narrow">

      <div style="text-align: right; position: fixed; left: calc(50% - 520px); padding-top: 0px; font-family: 'Roboto', sans-serif; font-weight: 300; font-size: 13px;">
        <div id='btn_computational' class='btn_info' onclick='show_help("computational");'><b>Computational</b></div><br>
      </div>

      <div id='div_computational'>
      <a id="computational"></a>

      <div class="div_header"><b>Computational Analysis</b></div>
      We provide pipelines for protocol-specific computational analysis of short and long reads. Currently, expressRNA supports Lexogen QuantSeq Reverse and Forward protocols, standard RNA-seq and Nanopore direct RNA. Please see below details of how we process reads from these protocols to compute a polyA database and count table.

      <br><br>
      <div class="div_header"><b><a href='https://www.lexogen.com/quantseq-3mrna-sequencing-rev/' target=_new>Lexogen 3' QuantSeq Reverse for Illumina</a></b></div>

      <div>We map (align) the reads to the reference genome with <a href="https://github.com/alexdobin/STAR" target=_new>STAR</a>. To <b>build the genome index</b>, we include GTF annotation files:
        <br><br>
        <div style='padding-left: 5px; padding-right: 5px; border-radius: 5px; background-color: #ceffc3; display: inline-block'>STAR --runMode genomeGenerate --genomeDir output_folder --genomeFastaFiles genome.fasta --sjdbGTFfile genome.gtf</div>
        <br><br>
      </div>

      After we align raw reads to the reference genome (default STAR settings) allowing soft clipping of bases and only reporting unique/best alignments are inclusive of at least 20% of the read length:

      <br><br>
      <div style='padding-left: 5px; padding-right: 5px; border-radius: 5px; background-color: #ceffc3; display: inline-block'>--outFilterMultimapNmax 1 --outFilterMatchNminOverLread 0.2 --outFilterScoreMinOverLread 0.2</div>

      <br>
      <br><img src='media/help/help_lexrev1.png' style='border-radius: 3%; padding: 2px; border:1px solid #e1e1e1; width:400px'><br>
      <div style='font-size: 12px'>The reads align in the opposite direction of genes (in the figure gene SMC1A on - strand and reads aligning to + strand, hg19).<br>The cleavage site is noted as the 5' of the aligned read.</div>
      <br>

      <div style='text-align:justify'>
      We then iterate over each alignment taking the first (5') nucleotide as a potential polyA cleavage site. We check the surrounding genomic sequence [-30..10] for A-rich regions. If this region contains either AAAAAAA or #A > 8, the alignment is filtered out. However, if we find a <font color=green class="btn" title="<div style='font-size:12px; font-family: Courier'>AATAAA<br>ATTAAA<br>TATAAA<br>AGTAAA<br>AATACA<br>CATAAA<br>AATATA<br>GATAAA<br>AATGAA<br>AAGAAA<br>ACTAAA<br>AATAGA<br>AATAAT<br>AACAAA<br>ATTACA<br>ATTATA<br>AACAAG<br>AATAAG</div>">polyA signal (PAS) / hexamer</font> in the region [-30..0] then the alignment is kept.
      <br><br>
      We then combine the remaining alignments and construct the <b>library polyA database</b>. Since cleavage and polyadenylation is not nucleotide exact, we sort all positional values (genome wide) from highest to lowest and perform a clustering of values in the range [-25..25]. We produce a table (bed graph) of the genomic loci and <b>annotate polyA sites</b> to genes with <b>5KB downstream extension</b> (or up to max half-distance to downstream gene).
      <br><br>
      To compute the polyA count table, we <b>take each genomic loci</b> in the library polyA database and <b>count the number of alignments</b> (5' nucleotide) from each experiment separately in the vicinity of the polyA site (<b>-5..5 nt</b>). Counting considers all alignments from each experiment in the library. Filtering is only applyed in the steps of constructing the library polyA database to account for <b>internal priming events</b>.

      </div>

      <br>
      <div class="div_header"><b><a href='https://www.lexogen.com/quantseq-3mrna-sequencing/' target=_new>Lexogen 3' QuantSeq Forward for Illumina</a></b></div>

      <br><br>
      <div class="div_header"><b>Alternative Polyadenylation (APA) Computational Analysis</b></div>

      expressRNA implements <b>alternative polyAdenylation (APA)</b> analysis from 3'-end short-read targeted sequence data and long-read Nanopore / PacBio data. In general the analysis is flexible to accomodate diverse sequence data, for example <a href="https://www.lexogen.com/quantseq-3mrna-sequencing/?gclid=CjwKCAjwhOvPBRBxEiwAx2nhLqPMeeMzqnWVmvDLAR3uSNW-YITTbG0UHBWCkiZk1w3wVdzd2ThL_BoCYCMQAvD_BwE" target=_lexogen>Lexogen Quantseq Forward and Reverse</a> protocols are well supported.

      The following steps are crucial for APA analysis:

      <div style="padding-left: 15px; padding-top: 10px;">
        <li> map / align the reads to the reference with STAR short read aligner
        <li> construct a polyA atlas / database by clustering alignment ends (or beginnings, depending on the protocol)
        <li> filter out alignments from A-rich genomic regions (to avoid internal priming events in the atlas construction step)
        <li> recover the signal by computing the "expression" of polyA sites by clustering
        <li> define comparative analysis (set of control experiments vs. set of test experiments)
        <li> run DEXSeq analysis on the polyA sites
        <li> select two sites per gene to classify polyA events (shortening, lenghtening of genes)
        <li> perform motif analysis around regulated sites with RNAmotifs2
        <li> Gene Ontology Enrichment Analysis on APA regulated genes
      </div>

      <img src="media/analysis_steps.png" width=700px>

      <br>
      The above list is just a set of automated analysis that can be performed on a set of experiments. Of course several alternatives can be implemented, such as using a pre-existing polyA atlas (database) for the analysis and computing the polyA site usage (counts) on the provided polyA site atlas.
      <br><br>
      Part of an APA report is a graphical representation of gene shortening and lenghtening. Our case study was an experiment of HEK-293 control vs. <font color=green class="btn" title="TDP-43 is a DNA and RNA-binding protein.<br><br><a target=_tdp43 href=https://en.wikipedia.org/wiki/TARDBP>Open wiki page</a> on TDP-43 to start exploring more.">TDP-43</font> KD. Repressed genes are blue (proximal polyA site is less used compared to the distal), and vice-versa for the enhanced genes. <a href="javascript:open_analysis('paper_final_all');">Open the analysis here</a>.
      <br><br>
      expressRNA provides the basic analysis in a completelly automatized fashion, however for research projects a lot of custom work still needs to be performed in terms of data quality control and pre-processing. Currently we are working to automatize all steps of analysis.

      </div>


</div>

<script src="help.js"></script>
