<div class="container">

    <div class="div_header"><b>News</b></div>

    <b><font color=#555555>29. August 2019</font></b><br>
    Added several genomes and the possibility to perform differential gene expression analysis. Working on the gene expression profile graphs.

    <br><br><img src="media/data_upload.svg" width=30 align="left" style='padding-right: 5px;'>
    Data upload is here! 1 library with up to 4 experiments (max 100 MB each) is available automatically (up to further notice). Scientists and researchers: <a href="mailto:gregor.rot@gmail.com?subject=expressRNA need more space">please simply write us</a> if you need more space for your data.

    <br><br>

    <b><font color=#555555>25. March 2019</font></b><br>
    <a href="mailto:gregor.rot@gmail.com?subject=expressRNA slack invite"><img src="media/slack.svg" width=60 align="left" style='padding-right: 5px; padding-top: 5px;'></a>
    We are using Slack to provide support and feedback for expressRNA analysis, you can join our channel by invite, simply <a href="mailto:gregor.rot@gmail.com?subject=expressRNA slack invite">write an e-mail</a> and let us know you would like to join.
    <br><br>

    <b><font color=#555555>22. February 2019</font></b><br>
    <img src="media/cloud.png" width=80 align="right">
    The expressRNA project is moving forward to provide <b>automatic data processing</b>.
    We plan to open test experiment data upload by end of February 2019. In this first stage, users will be able to upload experiments (FASTQ/FASTA) files and expressRNA will compute the polyA database, polyA expression profiles from the experimental data and additional global gene expression levels.
    <br><br>
    In terms of infrastructure, we are using a cloud server with 8-core CPU and 64 GB RAM, with enough disk space for this initial test of the data upload (scroll down to see <b>live server CPU usage, CLOUD</b>).
    <br><br>
    If you are logged into the system, you can notice an <a href='javascript:open_profile()'>updated profile section</a>, where you can see currently processed tasks / tickets.

    <br><br>

    <b><font color=#555555>16. January 2019</font></b><br>
    Several updates to the web interface, including user data upload and analysis.

    <br><br>

    <b><font color=#555555>20. July 2018</font></b><br>
    Moving fast forward to expressRNA v2, with user sequence data upload. We now have separate interfaces for Libraries (experiment datasets) and Analyses (alternative polyadenylation and RNAmotifs2). The interface is redesigned to be more user friendly, with search+pagination and other niceties. Top right <a href="https://github.com/grexor/expressRNA2/issues" target=_new>you can suggest ideas and report issues over GitHub</a> :-)

    <br><br>

    <b><font color=#555555>5. January 2018</font></b><br>
    Yey! Further developments and improvements on the web interface part. Now collaborators can search and browse analysis and libraries. It was about time, since we are adding one new library per week on average! Stay tuned for more.

    <br><br>

    <b><font color=#555555>22. December 2017</font></b><br>
    The alternative polyadenylation analysis interface is now refreshed, and we added a fold-change interactive graph to explore genes that undergo major proximal/distal switches. An example you can explore is our <a href="javascript:open_analysis('paper_final_all', 'High-resolution RNA maps demonstrate common principles of splicing and polyadenylation regulation by TDP-43');">published TDP-43 KD analysis</a>.

    <br><br>

    <b><font color=#555555>20. December 2017</font></b><br>
    We are working on the web interface redesign and upgrade, especially the search and browsing of analysis and sequencing libraries. The top menu first icon changed from "Browse" to <a href="javascript:open_search('analysis');">"Start"</a>, since with a single click you can access and browse analysis results.

    <br><br>

    <b><font color=#555555>15. December 2017</font></b><br>
    The main focus of expressRNA is in the processing of 3'-end targeted sequence data, watch the video to follow our analysis on the showcase of TDP-43 and it's role in APA.

    <br><br>

    <b><font color=#555555>15. November 2017</font></b><br>
    Exciting times for expressRNA in November, with several new ongoing collaborations and support for Nanopore and PacBio long-reads coming shortly. The long-reads spanning entire transcripts are ideal for the study of alternative polyadenylation and alternative splicing at the same time, and the team at expressRNA is developing a module to do just that. Stay tuned!

    <br><br>

    <b><font color=#555555>12. October 2017</font></b><br>
    expressRNA is undergoing extensive improvements and updates. Today we also moved to a new server. More news soon.

    <br><br>

    <b><font color=#555555>1. January 2017</font></font></b><br>
    The analysis platform is renamed to <b><font color=#555555>expressRNA</font></b>, since the platform combines several software for analysis of post-transcriptional processes, not only on the level of alternative polyadenylation.

    <br><br>

    <b><font color=#555555>25. June 2016</font></font></b><br>
    The alternative polyadenylation (APA) platform, apaExpress.org, is now online! In the preceding months, we applied the platform to several datasets and found it useful to share data and results between involved scientists. Now we are making it publicly available to serve the research community.

</div>
